include/knowhere/index.h

```diff
@@ -117,90 +122,84 @@ class Index {
+        span *= 0.000001;  // convert to s
+        kw_build_latency.Observe(span);
+#else
+        CHECK_STATUS(this->node->Build(dataset, *cfg));
+#endif
+        return Status::success;
     }
 
     Status
     Train(const DataSet& dataset, const Json& json) {
         Json json_(json);
         auto cfg = this->node->CreateConfig();
-        {
-            auto res = Config::FormatAndCheck(*cfg, json_);
-            LOG_KNOWHERE_DEBUG_ << "Train config dump: " << json_.dump();
-            if (res != Status::success) {
-                return res;
-            }
-        }
-        auto res = Config::Load(*cfg, json_, knowhere::TRAIN);
-        if (res != Status::success) {
-            return res;
-        }
-        return this->node->Train(dataset, *cfg);
+        CHECK_STATUS(Config::FormatAndCheck(*cfg, json_));
+        LOG_KNOWHERE_DEBUG_ << "Train config dump: " << json_.dump();
+        CHECK_STATUS(Config::Load(*cfg, json_, knowhere::TRAIN));
+
+        CHECK_STATUS(this->node->Train(dataset, *cfg));
+        return Status::success;
     }
 
     Status
     Add(const DataSet& dataset, const Json& json) {
         Json json_(json);
         auto cfg = this->node->CreateConfig();
-        {
-            auto res = Config::FormatAndCheck(*cfg, json_);
-            LOG_KNOWHERE_DEBUG_ << "Add config dump: " << json_.dump();
-            if (res != Status::success) {
-                return res;
-            }
-        }
-        auto res = Config::Load(*cfg, json_, knowhere::TRAIN);
-        if (res != Status::success) {
-            return res;
-        }
-        return this->node->Add(dataset, *cfg);
+        CHECK_STATUS(Config::FormatAndCheck(*cfg, json_));
+        LOG_KNOWHERE_DEBUG_ << "Add config dump: " << json_.dump();
+        CHECK_STATUS(Config::Load(*cfg, json_, knowhere::TRAIN));
+
+        CHECK_STATUS(this->node->Add(dataset, *cfg));
+        return Status::success;
     }
 
     expected<DataSetPtr, Status>
     Search(const DataSet& dataset, const Json& json, const BitsetView& bitset) const {
         Json json_(json);
         auto cfg = this->node->CreateConfig();
-        {
-            auto res = Config::FormatAndCheck(*cfg, json_);
-            LOG_KNOWHERE_DEBUG_ << "Search config dump: " << json_.dump();
-            if (res != Status::success) {
-                return unexpected(res);
-            }
-        }
-        auto res = Config::Load(*cfg, json_, knowhere::SEARCH);
-        if (res != Status::success) {
-            return unexpected(res);
-        }
-        return this->node->Search(dataset, *cfg, bitset);
+        CHECK_EXPECTED(Config::FormatAndCheck(*cfg, json_));
+        LOG_KNOWHERE_DEBUG_ << "Search config dump: " << json_.dump();
+        CHECK_EXPECTED(Config::Load(*cfg, json_, knowhere::SEARCH));
+
+#ifdef NOT_COMPILE_FOR_SWIG
+        TimeRecorder rc("Search");
+        auto res = this->node->Search(dataset, *cfg, bitset);
+        auto span = rc.ElapseFromBegin("done");
```

Comment:
Does this function print any message? Excessive messages may slow down search process

Reply:
TimeRecorder is generated with "log_level_ = 0", it will print nothing